src/main/java/io/tiledb/spark/TileDBPartitionReader.java

```diff
@@ -698,7 +707,14 @@ private void createValueVectors(long readBufferSize) throws TileDBError {
       RootAllocator allocator = ArrowUtils.rootAllocator();
       ArrowType arrowType;
       ValueVector valueVector;
-      ValueVector validityValueVector = new UInt1Vector(fieldName, allocator);
+
+      // In theory we could try to replace the following UInt2Vector with Uint1Vector. However,
+      // TileDB will throw an error that more validity cells are needed for the query. This
```

Comment:
I'm a little confused here, what is the error exactly?

Reply:
When allocating the size of the buffers for the query, apache arrow rounds up the size of the data buffer provided. For example, assuming we have an integer attribute and a read buffer size of 2000bytes. Arrow will round it up to 3000 bytes. (The numbers are just examples, I am not sure how the roundup occurs.) This makes the data buffers larger. When there is a nullable attribute the original size we provide ( (2000 bytes / 4) - one byte for each element) is not enough and TileDB throws an error requiring a larger validity map. Using a [Uint2Vector](https://arrow.apache.org/docs/java/reference/org/apache/arrow/vector/UInt2Vector.html) means that we use a validity map of size ( (2000 / 4 ) )  * 2  which is always enough.